### DISCUSSION TITLE
The provided export path doesn't match the page.

### DISCUSSION THREAD

**Discussion by suliskh** [vercel/next.js#16881] [Help] [answered]
Hi everyone 👋, I am new to Next. I have a page with dynamic routes as below:
- Homepage with a list of posts: `index.js`
- Detail page: `post/[slug].js`

When I run it on the dev server, its all work nicely. But when I try to run `next export`, it keeps showing:
`The provided export path '/post' doesn't match the '/post/[slug]' page.`

Is there any config I should set or anything else I am missing? 
Appreciate any help 🙏.

**Comment by DavidChouinard** [comment, marked as answer]
Where you able to resolve this? Having the same issue.

**Reply by suliskh** [reply]
I was having this problem because some `slug` is either `null` or `undefined`.
I fixed it by adding a default replacement if `slug` is either `null` or `undefined`:
```tsx
async function getStaticPaths() {
  // get posts with slug attribute
  let posts = await getPosts(); 

  // add "404" as default
  const paths = posts.map(({ slug }) => ({ params: { slug: slug || "404" } }));

  return {
    paths,
    fallback: false,
  };
}
```

I hope that helps 🙏🏻

**Reply by demonhunter3333** [reply]
That helped! Thanks!

**Reply by juanmcirco** [reply]
I want to say to you!!! Ilove you. You help me so much! jaja

**Comment by myriky** [comment]
any updates?

**Comment by zhouquan2016** [comment]
another way,make a index file:/post/index.js

import { useEffect } from "react";
import {useRouter} from "next/router"

export default function Page() {
    const router = useRouter();
    useEffect(() => {
        router.push("/post/1");
    }, []);
    return (<div></div>);
}
